component/src/main/java/io/siddhi/extension/execution/reorder/AlphaKSlackExtension.java

```diff
@@ -588,19 +564,82 @@ public Scheduler getScheduler() {
     @Override
     public void setScheduler(Scheduler scheduler) {
         this.scheduler = scheduler;
-        if (lastScheduledTimestamp < 0 && flag) {
-            lastScheduledTimestamp = siddhiAppContext.getTimestampGenerator().currentTime() +
-                    timerDuration;
-            scheduler.notifyAt(lastScheduledTimestamp);
-        }
     }
 
     private double calculateAlpha(double windowCoverageThreshold, double runtimeWindowCoverage) {
```

Comment:
Pass the sate to the calculateAlpha and you don't need to fetch that every time

Reply:
Fixed this in 2951106.